common/src/main/java/com/wynntils/overlays/StatusEffectsOverlay.java

```diff
@@ -207,5 +248,44 @@ public void setCount(int c) {
         public StatusEffect getEffect() {
             return this.effect;
         }
+
+        public void addModifier(StyledText modifier) {
+            this.modifierList.add(modifier.getStringWithoutFormatting());
+        }
+
+        private double extractDoubleFromString(String string) {
+            byte[] s = string.getBytes();
+            int len = string.length();
+
+            int start = 0;
+            while (!Character.isDigit(s[start]) && s[start] != '-') {
+                start += 1;
+            }
+
+            int end = start;
+            while (end < len && (Character.isDigit(s[end]) || s[end] == '.')) {
+                end += 1;
+            }
+            return Double.parseDouble(string.substring(start, end));
+        }
+
+        private int indexAfterDigits(String string) {
+            byte[] s = string.getBytes();
+            int len = string.length();
+            int i = 0;
+            while (!Character.isDigit(s[i])) {
+                i += 1;
+            }
+            while ( i < len && (Character.isDigit(s[i]) || s[i] == '.')) {
+                i += 1;
+            }
+            return i;
+        }
```

Comment:
I will most likely never accept anything with byte arrays.. We have so many tools available to us, there is no need to do this.

We might have had this conversation already but what's stopping you from adding a new capture group the the regex and parsing the number there?

Reply:
Also, I am just checking the previous addition, and I think even `MODIFIER_REGEX` is unnecessary. I need to login to get a grip on how this regex works though..

Reply:
The regex is in the `StatusEffectModel` class. So if we were to extract the number from there we could add a field to the `StatusEffect` class to hold the numerical value of the modifier.

We should also get the suffix of the modifier then, this can probably be done in the `RenderedStatusEffect` class by searching for either a '%' or a '/' (in /3s modifiers).

Just thinking out loud. What do you think of that approach?

Reply:
Why not put the suffix part in StatusEffect as well?

Reply:
That works too. I thought RenderedStatusEffect would be a better place since its only used in 1 of the 3 options